ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml

```diff
@@ -342,6 +356,240 @@ objects:
           oc wait namespace/openshift-ssh-bastion --for delete --timeout=10m
         }
 
+        function recover-from-etcd-quorum-loss() {
+          mkdir -p ~/.ssh || true
+          cp "${KUBE_SSH_KEY_PATH}" ~/.ssh/id_rsa
+          chmod 0600 ~/.ssh/id_rsa
+          echo "${USER:-default}:x:$(id -u):$(id -g):Default User:$HOME:/sbin/nologin" >> /etc/passwd
+
+          echo "Setting up ssh bastion"
+          curl https://raw.githubusercontent.com/eparis/ssh-bastion/master/deploy/deploy.sh | bash
+          for i in $(seq 0 60)
+          do
+            BASTION_HOST=$(oc get service -n openshift-ssh-bastion ssh-bastion -o jsonpath='{.status.loadBalancer.ingress[0].hostname}')
+            if [ ! -z "${BASTION_HOST}" ]; then break; fi
+            sleep 10
+          done
+
+          function bastion_ssh() {
+            retry 10 \
+              ssh -o LogLevel=error -o ConnectionAttempts=100 -o ConnectTimeout=5 -o StrictHostKeyChecking=no \
+                  -o ProxyCommand="ssh -A -o StrictHostKeyChecking=no -o LogLevel=error -o ServerAliveInterval=30 -o ConnectionAttempts=100 -o ConnectTimeout=5 -W %h:%p core@${BASTION_HOST}" \
+                  $@
+          }
+
+          # Machine API won't let the user to destroy the node which runs the controller
```

Comment:
Is there a bug for this?  Seems like a bug.  Doesn't mean we can't work around it here, but I don't want to forget about fixing it ;).

Reply:
https://bugzilla.redhat.com/show_bug.cgi?id=1700931 - this is a limitation of Machine API